### DISCUSSION TITLE
Accessing relational fields in a custom interface extension.

### DISCUSSION THREAD

**Discussion by mark-james** [directus/directus#7481] [Archive]
Hi @nickrum,
When creating a custom interface for an O2M relation, what is the best way of accessing the child items data?

I can see I can access the Primary Keys via the value prop. But what is the best way of then getting the rows?

Are there internal libraries available that I can hook into?

Thanks
Mark

**Comment by IlyaPetrovM** [comment]
Your question is similar to this: https://github.com/directus/directus/discussions/7457

**Reply by mark-james** [reply]
Thanks @IlyaPetrovM. Quite right, it is.

**Reply by mark-james** [reply]
Hi @IlyaPetrovM ,
To clarify based on your findings ,
adding 
```	
fields: () => {
	return ['id', 'statement', 'type']
},
```
to my index.js should allow me to access these field values via the value prop in my vue component? Or via another prop?

My value prop is still only showing me an array of primary keys after adding the above.

Thanks
Mark

**Reply by IlyaPetrovM** [reply]
You should set `value` prop type to `Object` and use mentioned fields as keys in template.

in example:
### src/display.vue
```js
<template>
    <span > 
        {{ value['id'] }} ,
        {{ value['statement'] }}
    </span>
</template>


<script>
    export default {
        props: {
            value: Object,
        },

    };
</script>
```

**Reply by mark-james** [reply]
Hi @nickrum, are you able to confirm if the fields prop is available to custom interfaces? Or is it just available for custom displays? I'm not able to reproduce the above for an interface.

**Reply by nickrum** [reply]
The `fields` property is indeed only available for custom displays. To access relational data inside an interface, you have to use the relations store as far as I know.

**Reply by mark-james** [reply]
Thanks @nickrum. How do I do that?

Can you point me in the direction of an example?

**Reply by nickrum** [reply]
You could take a look at how the "One to Many" interface in `app/src/interfaces/list-o2m` does it. Stores can be accessed by injecting the `system` property and then using `this.system.useRelationsStore()`.

**Reply by mark-james** [reply]
Sorry if this is obvious, but how do I inject the system property? 

Is this an undocumented prop available in the vue interface component?

**Reply by nickrum** [reply]
There is an example in the [module docs](https://docs.directus.io/guides/modules/#accessing-the-api-from-within-your-extension). You basically just add `inject: ['system']` to your interface component.
